src/Server.fs

```diff
@@ -612,10 +668,13 @@ type LspServer() =
 
   /// The workspace symbol request is sent from the client to the server to list project-wide symbols matching
   /// the query string.
-  abstract member WorkspaceSymbol: WorkspaceSymbolParams -> AsyncLspResult<SymbolInformation [] option>
+  abstract member WorkspaceSymbol: WorkspaceSymbolParams -> AsyncLspResult<U2<SymbolInformation [], WorkspaceSymbol []> option>
```

Comment:
It'll write just fine, but on the read side it really only supports unions where the cases are distinct combinations of number, bool, string, and some object - it can't reliably read any more distinct than that currently.

So for this specifically it may be ok since the erased union is on the sending side, but it's worth keeping an eye out. We may want to add an explicit throw to the ReadJson member if a destination type invalidates this invariant.